### ISSUE TITLE
What does the interleave method do

### ISSUE THREAD

**Issue by atinghosh** [YU1ut/MixMatch-pytorch#20]
Thanks for your code. I was trying to implement mixmatch independently but has been unable to achieve accuracy as mentioned in the paper. 

I omitted that part and I noticed no difference. 

Can you please elaborate on what this method does. Thanks a lot.

**Comment by berzentine** [comment]
Why do we need this method to get the correct BatchNorm? I m confused too. Any insights will be very helpful

**Comment by atinghosh** [comment]
@berzentine I should have closed this issue long back, I retract what I mentioned earlier, actually without the interleave you won't get high accuracy. 

Interleave makes sure batch norm statistic is calculated correctly for the small labelled dataset, each time the forward propagation is done BatchNorm (BN) keeps track of the mean and standard deviation of minibatch of the activations, and later at the time of prediction/inference these stored means and standard deviations (probably some exponentially moving averaged version of them) are used. Now, if, one keeps forwarding the small labelled dataset through NN, again and again, these BN parameters get biased. Interleave method avoid that. 

I am closing this issue.
